src/main/java/frc/robot/subsystems/ColorSensor.java

```diff
@@ -0,0 +1,55 @@
+package frc.robot.subsystems;
+import com.revrobotics.ColorSensorV3;
+import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.util.Color;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorMatchResult;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.utils.smartshuffleboard.SmartShuffleboard;
+
+public class ColorSensor extends SubsystemBase{
+    private final I2C.Port i2cPort = I2C.Port.kOnboard;
+    private final ColorSensorV3 m_colorSensor = new ColorSensorV3(i2cPort);
+    private final ColorMatch m_colorMatcher = new ColorMatch();
+    String colorObject;
+    public ColorSensor() {
+        m_colorMatcher.addColorMatch(Constants.K_PIECE_TARGET);
+        m_colorMatcher.addColorMatch(Constants.K_PLASTIC_TARGET);
+    }
+    @Override
+    public void periodic() {
+        Color detectedColor = m_colorSensor.getColor();
+        double IR = m_colorSensor.getIR();
+        SmartShuffleboard.put("Feeder","Red", detectedColor.red);
```

Comment:
Please create a new constant in GameConstants, "COLOR_SENSOR_DEBUG", add all the SmartShuffleboard items into an "if (Constants.COLOR_SENSOR_DEBUG)" section.